### PR TITLE
Add a PR template with a checklist.

### PR DESCRIPTION
**Pull request by qingling128**

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+- Unit tests
+  - [ ] Unit tests do not apply.
+  - [ ] Unit tests have been added/modified and passed for this PR.
+- Integration tests
+  - [ ] Integration tests do not apply.
+  - [ ] Integration tests have been added/modified and passed for this PR.
+- Documentation
+  - [ ] This PR introduces no user visible changes.
+  - [ ] This PR introduces 3rd party app support change, and the related doc changes have been made in the `metadata.yaml` files.
+  - [ ] This PR introduces user visible changes that require direct collaboration with tech writers in the internal code base separately.
```
Comment (sophieyfang) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:21]: one such case I can imagine is the mssql is v2 metrics. It needs manual handling in the doc contents. It's beyond current metadata files can suffice. 

Maybe we shall make it sound clear in this sentence

Comment (qingling128) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Sync'd with Sophie offline. We decided to use this checklist just as a reminder, and defer the details of documentation requirements to the developer and reviewer, because it varies case by case a lot.

I have pushed a commit to simplify the wording.

**Comment by qingling128** [comment]
Ready for a second look

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+## Description
+<!--- Describe your changes in detail. -->
+
+## Related issue
+<!--- Add a link to the issue (follow the b/XXX format for internal issues) -->
+
+## How has this been tested?
+<!--- Please describe how you tested the changes besides the automatically triggered unit tests when applicable. -->
+<!--- Include sample output logs or metrics and/or screenshots of key results when applicable. -->
```
Comment (sophieyfang) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:9]: Will it be too strong to say "Must Include"?

Comment (qingling128) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Done

**Review by sophieyfang — APPROVED** [review]
